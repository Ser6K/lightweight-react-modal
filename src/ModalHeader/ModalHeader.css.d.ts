declare namespace ModalHeaderCssNamespace {
  export interface IModalHeaderCss {
    header: string;
  }
}

declare const ModalHeaderCssModule: ModalHeaderCssNamespace.IModalHeaderCss & {
  /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
  locals: ModalHeaderCssNamespace.IModalHeaderCss;
};

export = ModalHeaderCssModule;
