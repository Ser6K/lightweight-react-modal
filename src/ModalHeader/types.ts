import { ClassNameType, ReactNodeType } from 'src/types'

export interface ModalHeaderTypes {
  children: ReactNodeType
  className?: ClassNameType
  [propName: string]: any
}
