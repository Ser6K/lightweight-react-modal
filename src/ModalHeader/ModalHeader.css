.header {
    flex-shrink: 0;
    border-bottom: 1px solid grey;
    padding: 15px 20px;
    box-sizing: border-box;
}