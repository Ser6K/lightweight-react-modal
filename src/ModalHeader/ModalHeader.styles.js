import styled from 'styled-components';

export const Header = styled.div`
    flex-shrink: 0;
    display: flex;
    align-items: center;
    justify-content: space-between;
    border-bottom: 1px solid grey;
    padding: 15px 20px;
    box-sizing: border-box;
`;
