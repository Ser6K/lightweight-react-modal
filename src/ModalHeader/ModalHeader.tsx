import React from 'react'
import { classnames } from 'src/utils'

import { ModalHeaderTypes } from './types'

import styles from './ModalHeader.css'

const ModalHeader: React.FC<ModalHeaderTypes> = ({ children, className, ...props }) => (
  <div {...props} className={classnames(styles.header, className)}>
    {children}
  </div>
)

export default ModalHeader
