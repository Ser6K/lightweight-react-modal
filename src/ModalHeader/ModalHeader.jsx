import React from 'react';
import PropTypes from 'prop-types';
import { HeaderDisplayName } from 'utils';
import { Header } from './ModalHeader.styles';

const ModalHeader = React.memo(({ children, ...props }) => (
    <Header {...props}>
        {children}
    </Header>
));

ModalHeader.propTypes = {
    children: PropTypes.node.isRequired,
};

ModalHeader.displayName = HeaderDisplayName;
export default ModalHeader;
