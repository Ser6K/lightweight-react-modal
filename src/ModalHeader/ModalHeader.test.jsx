import React from 'react';
import { mount } from 'enzyme';
import ModalHeader from './ModalHeader';

describe('Modal header', () => {
    test('it matches snapshot with no props', () => {
        const modalHeader = mount(<ModalHeader>Header</ModalHeader>);
        expect(modalHeader).toMatchSnapshot();
    });
    test('it matches snapshot with props', () => {
        const modalHeader = mount(<ModalHeader className="className" style={{ display: 'inline' }}>Header</ModalHeader>);
        expect(modalHeader).toMatchSnapshot();
    });
});
