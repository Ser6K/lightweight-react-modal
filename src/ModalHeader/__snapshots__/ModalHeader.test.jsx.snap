// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal header it matches snapshot with no props 1`] = `
<ModalHeader>
  <ModalHeaderstyles__Header>
    <div
      className="ModalHeaderstyles__Header-sc-1f4o000-0 hfQxOK"
    >
      Header
    </div>
  </ModalHeaderstyles__Header>
</ModalHeader>
`;

exports[`Modal header it matches snapshot with props 1`] = `
<ModalHeader
  className="className"
  style={
    Object {
      "display": "inline",
    }
  }
>
  <ModalHeaderstyles__Header
    className="className"
    style={
      Object {
        "display": "inline",
      }
    }
  >
    <div
      className="ModalHeaderstyles__Header-sc-1f4o000-0 hfQxOK className"
      style={
        Object {
          "display": "inline",
        }
      }
    >
      Header
    </div>
  </ModalHeaderstyles__Header>
</ModalHeader>
`;
